packages/components/src/internal/components/editable/LookupCell.tsx

```diff
@@ -114,6 +114,12 @@ export class LookupCell extends PureComponent<LookupCellProps> {
             queryFilters = queryFilters.push(...lookup.getQueryFilters(operation));
         }
 
+        let selectValue = isMultiple ? rawValues : rawValues[0];
+        // Issue 49502: If the lookup is a measurement unit, then we need to use the unit's display value
+        if (col.isUnitsLookup()) {
+            selectValue = values.find(vd => vd.display !== undefined)?.display;
```

Comment:
Kind of weird that this uses a find. There's only one value possible, right?

Reply:
Yeah...I was just matching what this component is doing for the `rawValues` const. But for the units lookup case, the array would only ever have one element. So I agree that find is a little weird.